mm/mm_heap/mm_memalign.c

```diff
@@ -227,5 +230,7 @@ FAR void *mm_memalign(FAR struct mm_heap_s *heap, size_t alignment,
   kasan_unpoison((FAR void *)alignedchunk,
                  mm_malloc_size((FAR void *)alignedchunk));
 
-  return (FAR void *)alignedchunk;
+  ptr = (FAR void *)alignedchunk;
+  DEBUGASSERT(ptr == NULL || ((uintptr_t)ptr) % alignment == 0);
+  return ptr;
```

Comment:
Maybe just
```suggestion
  DEBUGASSERT(alignedchunk % alignment == 0);
  return (FAR void *)alignedchunk;
```
to reduce number of lines in the PR.

Reply:
i'm not sure what's the point of the suggested change.
is it that important to save a line? why?

Reply:
1. `0 % alignment  == 0` so we can eliminate `ptr == NULL` check
2. we can reuse existing variables and not introduce a new one.

Reply:
anyway I wrote that all suggestions are optional and we can keep the code like it is now.

Reply:
i prefer not to do it because bitwise operations on NULL is evil.

Reply:
`alignedchunk` -- is `size_t` so not NULL based arithmetic operations

Reply:
We can even update `alignedchunk` from `size_t` to `uintptr_t` to make it even more clear.

Reply:
i applied this one as it can't be NULL here.